include/openthread/nat64.h

```diff
@@ -87,6 +87,49 @@ typedef struct otIp4Cidr
     uint8_t      mLength;
 } otIp4Cidr;
 
+/**
+ * Allocate a new message buffer for sending an IPv4 message (which will be translated into an IPv6 packet by NAT64
+ * later). Message buffers allocated by this function will have 20 bytes (The differences between the size of IPv6
+ * headers and the size of IPv4 headers) reserved.
+ *
+ * This function is available only when OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE is enabled.
+ *
+ * @note If @p aSettings is 'NULL', the link layer security is enabled and the message priority is set to
+ * OT_MESSAGE_PRIORITY_NORMAL by default.
+ *
+ * @param[in]  aInstance  A pointer to an OpenThread instance.
+ * @param[in]  aSettings  A pointer to the message settings or NULL to set default settings.
+ *
+ * @returns A pointer to the message buffer or NULL if no message buffers are available or parameters are invalid.
+ *
+ * @sa otMessageFree
+ * @sa otBorderRouterSend
+ *
+ */
+otMessage *otIp4NewMessage(otInstance *aInstance, const otMessageSettings *aSettings);
```

Comment:
I think the typing cannot prevent the user from mistakenly fill a buffer by wrong payload.

The border routers (and for most border routers) that are just simple packet forwarders (like the one in posix/platform/netif.cpp), may want to use the same function for both IPv4 and IPv6 messages. Having two functions makes it inconvenient.

I agree that abusing Ip6 type for general IP is not a good idea.

Reply:
Correct, user can still fill wrong payload. 

We can keep it simple and not have `otIp4NewMessage()` (require user to always use `otIp6NewMessage()` but then the reserved header won't be correct).

Or if we have `oIp4tMessage *otIp4NewMessage()` and then define an OT API `otNat64SendMessage(otInstance *aInstance, otIp4Message *aIp4Message)`, then user can only pass it a `otIp4Message` which it can only get from the `otIp4NewMessage()` (user cannot pass a `otMessage` to this API anymore).

Reply:
Updated

I also added the `otIp4MessageFree` for releasing the message.

I also provided an otCastIp4Message so the users can still use functions like `otMessageAppend`, `otMessageRead`, that makes it possible to share as much code as possible (in netif.cpp).

On the receiving side, I provided `otNat64SetReceiveIp4Callback`, the implementation lives in the Ip6 module. And the callback receives an `otIp4Message` to make it consistent.

Reply:
Thanks. 👍